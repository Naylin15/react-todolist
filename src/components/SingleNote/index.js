import React from 'react';
import EditNote from '../EditNote';
import { Checkbox, Grid, Header, Dropdown } from 'semantic-ui-react';
import DropdownCategory from '../DropdownCategory';

class SingleNote extends React.Component {

    render() {
        const {note, noteId, removeNote, saveChanges, editNote, checkNote} = this.props;

        if (note.edit !== true) {
            // it's undefined but it should be false
            return (
                <Grid.Row centered columns={3} id={noteId}>
                    <Grid.Column width={2}>
                        <Checkbox onChange={(e, data) => checkNote(noteId, e, data)}/>
                    </Grid.Column>
                    <Grid.Column>
                        <Header as='h2' id={'note-' + noteId}>{note.title}</Header>
                    </Grid.Column>
                    <Grid.Column>
                        <DropdownCategory />
                        <Dropdown icon='ellipsis vertical'>
                            <Dropdown.Menu>
                                <Dropdown.Item text='Delete' onClick={() => removeNote(noteId)}/>
                                <Dropdown.Item text='Edit' onClick={() => editNote(noteId)} />
                            </Dropdown.Menu>
                        </Dropdown>
                    </Grid.Column>
                </Grid.Row>
            )

        } else {
            return(
                <EditNote
                    note={note}
                    noteId={noteId}
                    saveChanges={saveChanges}
                />
            )
        }    
    }
}

export default SingleNote;