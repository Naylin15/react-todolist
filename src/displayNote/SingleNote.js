import React from 'react';
import { Checkbox, Grid, Header, Dropdown } from 'semantic-ui-react';
import DropdownCategory from './DropdownCategory';


class SingleNote extends React.Component {
    render() {
        const {note, removeNote, noteId} = this.props;

        return(
            <Grid.Row centered columns={3} >
                <Grid.Column width={2}>
                    <Checkbox />
                </Grid.Column>
                <Grid.Column>
                    <Header as='h2'>{note.title}</Header>
                </Grid.Column>
                <Grid.Column>
                    <DropdownCategory />
                    <Dropdown icon='ellipsis vertical'>
                        <Dropdown.Menu>
                            <Dropdown.Item text='Delete' onClick={() => removeNote(noteId)}/>
                            <Dropdown.Item text='Edit' />
                        </Dropdown.Menu>
                    </Dropdown>
                </Grid.Column>
            </Grid.Row>
        )

    }

}

export default SingleNote;