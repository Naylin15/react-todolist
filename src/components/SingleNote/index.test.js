import React from 'react';
import renderer from 'react-test-renderer'
import SingleNote from './index';

it('renders correctly', () => {
    const tree = renderer
      .create(<SingleNote note={{edit: false, id: 0, title: '123'}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });