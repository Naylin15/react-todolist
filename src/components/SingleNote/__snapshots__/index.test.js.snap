// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="centered three column row"
>
  <div
    className="two wide column"
  >
    <div
      className="ui fitted checkbox"
      onChange={[Function]}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <input
        checked={false}
        className="hidden"
        readOnly={true}
        tabIndex={0}
        type="checkbox"
      />
      <label />
    </div>
  </div>
  <div
    className="column"
  >
    <h2
      className="ui header"
      id="note-undefined"
    >
      123
    </h2>
  </div>
  <div
    className="column"
  >
    <div
      aria-expanded={false}
      className="ui dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <i
        aria-hidden="true"
        className="filter icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          <span
            className="text"
          >
            Important
          </span>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          <span
            className="text"
          >
            Tomorrow
          </span>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          <span
            className="text"
          >
            Later
          </span>
        </div>
      </div>
    </div>
    <div
      aria-expanded={false}
      className="ui dropdown"
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="listbox"
      tabIndex={0}
    >
      <i
        aria-hidden="true"
        className="ellipsis vertical icon"
        onClick={[Function]}
      />
      <div
        className="menu transition"
      >
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          <span
            className="text"
          >
            Delete
          </span>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="option"
        >
          <span
            className="text"
          >
            Edit
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
